src/frequency_manager.cpp

```diff
@@ -0,0 +1,30 @@
+#include "real_time_tools/frequency_manager.hpp"
+
+#include <iostream>
+
+namespace real_time_tools
+{
+FrequencyManager::FrequencyManager(double frequency)
+    : period_ms_((1.0 / frequency) * 1000.0), previous_time_ms_(-1)
+{
+}
+
+bool FrequencyManager::wait()
+{
+    double t = Timer::get_current_time_ms();
```

Comment:
Naive question: why not using wait until?

Reply:
Not sure what you mean. Do you mean: calling the function "wait_until" ?